File: src/main/java/factory_method/creator/UltraVehicleFactory.java
package factory_method.creator;

import factory_method.product.UltraBus;
import factory_method.product.UltraTruck;
import factory_method.product.Vehicle;
import factory_method.product.VehicleType;

public class UltraVehicleFactory extends VehicleFactory {
    @Override
    protected Vehicle createVehicle(VehicleType vehicleType) {
        return switch (vehicleType) {
            case BUS -> new UltraBus();
            case TRUCK -> new UltraTruck();
        };
    }
}

File: src/main/kotlin/factory_method/creator/UltraVehicleFactory.kt
package factory_method.creator

import factory_method.product.UltraBus
import factory_method.product.UltraTruck
import factory_method.product.Vehicle
import factory_method.product.VehicleType

class UltraVehicleFactory : VehicleFactory() {
    override fun createVehicle(vehicleType: VehicleType): Vehicle {
        return when (vehicleType) {
            VehicleType.BUS -> UltraBus()
            VehicleType.TRUCK -> UltraTruck()
        }
    }
}